### PR TITLE
Add new endpoint - setHRInterview

### PR DESCRIPTION
**Pull request by AnnKirienko**

### CODE REVIEW

File: BLL/DTO/HRInterviewDTOWithDecline.cs [hunk 1]
```diff
@@ -8,6 +8,6 @@ namespace BLL.DTO
 {
   public  class HRInterviewDTOWithDecline : HRInterviewDTO
```
Comment (DShcharbakou) [inline BLL/DTO/HRInterviewDTOWithDecline.cs:9]: would be better call this method like HRInterviewDTOWithStatus

File: BLL/Interfaces/ISkillKnowledgeService.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using BLL.DTO;
+using System.Collections.Generic;
+
+namespace BLL.Interfaces
+{
+    public interface ISkillKnowledgeService
+    {
+      //  IEnumerable<SkillKnowledgeDTO> GetList();
```
Comment (DShcharbakou) [inline BLL/Interfaces/ISkillKnowledgeService.cs:8]: delete this

File: BLL/MappingProfiles/MappingProfile.cs [hunk 3]
```diff
@@ -15,8 +15,8 @@ public class MappingProfile : Profile
         public MappingProfile()
         {
             CreateMap<Candidate, CandidateDTO>();
-              /*  .ForMember(cd => cd.ID,
-                 cd => cd.MapFrom(cd => cd.Id);*/
+            /*  .ForMember(cd => cd.ID,
+               cd => cd.MapFrom(cd => cd.Id);*/
```
Comment (DShcharbakou) [inline BLL/MappingProfiles/MappingProfile.cs:19]: delete this

File: BLL/Services/InterviewService.cs [hunk 4]
```diff
@@ -42,11 +43,22 @@ public IEnumerable<InterviewDTO> GetList()
...
             _db.Save();
+            return interview.Id;
+        }
+        public int AddHRInterviewWithDecline(HRInterviewDTOWithDecline hrInterviewDTOWithDecline)
+        {
+            var interview = _mapper.Map<Interview>(hrInterviewDTOWithDecline);
+            _db.Interviews.Save(interview);
+            //var candSand = _db.Interviews.FindWithSpecificationPattern(new InterviewStatusSpecification()).FirstOrDefault(x => x.CandidateID == interview.CandidateID);
+            //var cand = candSand.Candidate.CandidateSandboxes.Where(x => x.Status.Id == hrInterviewDTOWithDecline.StatusID);
+            //hrInterviewDTOWithDecline.StatusID
```
Comment (DShcharbakou) [inline BLL/Services/InterviewService.cs:59]: delete this

File: BLL/Services/SkillKnowledgeService.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        private readonly IUnitOfWork _db;
+        private readonly IMapper _mapper;
+        public SkillKnowledgeService(IUnitOfWork db, IMapper mapper)
+        {
+            _db = db;
+            _mapper = mapper;
+        }
+        public void AddSkillKnowledge(IEnumerable<SkillKnowledgeDTO> skillKnowledgeDtoList)
+        {
+            foreach (var skillKnowledgeDto in skillKnowledgeDtoList)
```
Comment (DShcharbakou) [inline BLL/Services/SkillKnowledgeService.cs:26]: bad practice. Save into database in the loop

File: BLL/Services/SkillKnowledgeService.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        }
+
+
+        /* 
+                public IEnumerable<SkillKnowledgeDTO> GetList()
+                {
+                    return _mapper.Map<IEnumerable<SkillKnowledge>, List<SkillKnowledgeDTO>>(_db.SkillKnowledges.GetAll());
+                }     
+              */
```
Comment (DShcharbakou) [inline BLL/Services/SkillKnowledgeService.cs:50]: delete this

File: UI/Controllers/HRController.cs [hunk 7]
```diff
@@ -49,17 +52,29 @@ public CandidateDTO Get(int id)
...
+        {
+            HRInterviewDTOWithDecline hRInterviewWithDecline = _mapper.Map<HRInterviewDTOWithDecline>(hrInterviewDTODecline);
+            var employee = await GetEmployee();
+            hRInterviewWithDecline.EmployeeID = employee.Id;
+            //hRInterviewWithDecline.StatusID
+            hRInterviewWithDecline.ID = _interviewService.AddHRInterview(hRInterviewWithDecline);
+
+            var skillKnowledgeDTOList = _mapper.Map<IEnumerable<SkillKnowledgeDTO>>(hRInterviewWithDecline);
+            _skillKnowledgeService.AddSkillKnowledge(skillKnowledgeDTOList);
+        }
```
Comment (DShcharbakou) [inline UI/Controllers/HRController.cs:77]: I don't see differences between InterviewResults and InterviewResultsWithDeclineStatus

Comment (AnnKirienko) [reply UI/Controllers/HRController.cs:77]: I am too, but I don't know how I can use endpoint method

**Review by MikalaiKusakin — APPROVED** [review]

**Review by Alexy-Def — APPROVED** [review]

File: BLL/Services/SkillKnowledgeService.cs (hunk 5 above)
Comment (AnnKirienko) [reply BLL/Services/SkillKnowledgeService.cs:26]: we can remake this in future

**Review by DShcharbakou — APPROVED** [review]

**Review by CatholicCardinal — APPROVED** [review]

**Pull request merged**